src/core/lib/iomgr/cfstream_handle.cc

```diff
@@ -68,8 +69,10 @@ void CFStreamHandle::ReadCallback(CFReadStreamRef stream,
       handle->read_event_.SetReady();
       break;
     case kCFStreamEventErrorOccurred:
-      handle->open_event_.SetReady();
-      handle->read_event_.SetReady();
+      error = GRPC_ERROR_CREATE_FROM_STATIC_STRING("Read error");
```

Comment:
I would suggest getting the error detail here rather than just a "read error". See the corresponding way that error is handled in `ReadAction` and `WriteAction`.

Reply:
Thanks for pointing that out, fixed.

Reply:
Don't forget `CFStreamAnnotateError` too :)

Reply:
Added call to `grpc_error_set_int()`, as we don't have to CFStreamEndpoint here.